### ISSUE TITLE
 Issues with celery work pidbox and connection reset by peer for jobs over 2 minutes run time 

### ISSUE THREAD

**Issue by mneumarkidg** [ansible/awx#1188]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - UI

##### SUMMARY
UI shows job failed status even when worker in background is still running in background. Gives users false status.

##### ENVIRONMENT
See Google Posting for more: https://groups.google.com/forum/#!topic/awx-project/fjKjQa78f_A

##### STEPS TO REPRODUCE

Running a job that takes > than 5 minutes to execute

##### EXPECTED RESULTS

Running status instead of failed. 

##### ACTUAL RESULTS

Failed Status.

Process is still running in background and later will update the correct status, however shows job failure until correct status update.

##### ADDITIONAL INFORMATION

See Google Posting for more: https://groups.google.com/forum/#!topic/awx-project/fjKjQa78f_A

**Comment by wenottingham** [comment]
Please retry with https://github.com/ansible/awx/pull/1185, that may fix this.

**Comment by mneumarkidg** [comment]
@wenottingham I just tried #1185 and noticed that if the job is scheduled now it doesn't show that error, if the job is manually executed then it will show that error in the UI. Not sure if this helps?

**Comment by mneumarkidg** [comment]
Appears to be manual jobs show failure of celery if running over 5 minutes

**Comment by wenottingham** [comment]
Moving to API, all the info in the UI comes from the API.

**Comment by mneumarkidg** [comment]
Upgraded to "1.0.4.33" today and still having the same issues with celery workers. Only difference is the error changed due to source code changes.

`Task was marked as running in Tower but was not present in the job queue, so it has been marked as failed. `

Task shows failure when running > 5 minutes, will update correct status upon job completetion. However job status is incorrect (showing failed) until job completes. Making uses think the job failed.

**Comment by mneumarkidg** [comment]
![screen shot 2018-02-28 at 2 24 43 pm](https://user-images.githubusercontent.com/25063770/36816619-2f2ad00a-1c93-11e8-9396-18e9af5e76a4.png)

**Comment by mneumarkidg** [comment]
Same job after job runs
![screen shot 2018-02-28 at 2 31 44 pm](https://user-images.githubusercontent.com/25063770/36816892-2238c07c-1c94-11e8-9a89-4d32de9bb5a7.png)

**Comment by grahamneville** [comment]
I was concerned this was going to affect me on an upgrade so thought I'd test it out. I'm unable to replicate this issue:

I manually executed a job that has a 15min pause, I can see the job running at all times, even after browsing to different parts of AWX and then coming back:

![image](https://user-images.githubusercontent.com/19539695/36866896-8049fba2-1d8b-11e8-9226-e95dca1c2cec.png)

The job completes successfully after 15mins:

![image](https://user-images.githubusercontent.com/19539695/36866964-b3b8e5c0-1d8b-11e8-8c34-a35734131661.png)

I'm running a clean install of 1.0.4.33.

**Comment by mneumarkidg** [comment]
@grahamneville What setup are you using? I'm just using local docker with two instances configured in the tower instance group.

**Comment by mneumarkidg** [comment]
The jobs even trigger "Job #10550 had status failed" email to be sent out to the users. However, job #10550 finished successfully. Just shows failed during job running and send out failure emails to all the users. Which are false positives since the job is still running.

**Comment by grahamneville** [comment]
I'm not using instances, just local docker and one of each of the containers.

**Comment by matburt** [comment]
@mneumarkidg How did you deploy this? The standalone docker only supports a single instance.

**Comment by matburt** [comment]
This is a classic split brained mode that you can get in if the awx instance's celery queues aren't working together... the systems don't know what the other is working on and can't resolve the state with what's in the database. The healthcheck time for jobs is ~2m

This is why it's important to use our official install methods.

**Comment by matburt** [comment]
Closing after discussing with mneumarkidg

**Comment by mmontesi** [comment]
Hi all,
I'm having the same issue on a kubernetes installation (on Azure AKS).
My workflows worked fine till yesterday, when appeared this behaviour.
I had only added 2 users (still not used by anyone) and a fake notification that I removed.
Now workflows fail because each job is marked as failed when is simply running.
Can you help me? @matburt what did you discuss with @mneumarkidg before marking this issue as closed?
I understand it's something related with queues but I can't imagine what went wrong and (more important) how to fix it.
Let me know for more infos you need to investigate.
Thanks in advance,
Marco

**Comment by ryanpetrello** [comment]
@mmontesi what version of AWX are you using?

If you're still seeing celery pidbox errors, you may want to try upgrading to a newer version - we removed our usage of the celery library last year due to this and other issues.

**Comment by mmontesi** [comment]
@ryanpetrello I was on 2.1.2 when I first experienced the issue, and till yesterday the same job templates have worked regularly for weeks.
Then I downloaded 3.0.1 zip and upgraded in place: same behaviour.
In site upgrade seems to work but official Red Hat's FAQ says it's not supported.
Tomorrow at work I'll try on a fresh 3.0.1 installation.
The issue came up in a staging environment, in which I didn't manage to have backups (damn me), so I can't roll back to a working point to insulate the possible cause.
I'd like to understand what went wrong because I'm planning to move in production in few weeks.
Is there anything I have to check? Logs? How can I verify queues are working as expected?

UPDATE this morning workflows failing last friday are working again @-@
Is it possible scheduled management jobs have fixed the issue? I started these jobs manually but with no effect, two days after all is ok, this is really strange.
